### PR TITLE
Clear votes which were created before spork15 activation and use operator key for non-funding votes

### PR DESCRIPTION
**Pull request by codablock**
This will purge all votes which were created before spork15 activation. This means that all MNOs will have to re-vote for all proposals after spork15 activation.

The reason for this is that the introduction of separate voting keys in DIP3 invalidates all votes on activation of spork15.

This PR also fixes an issue with deletion votes from sentinel. The old code enforced validation against the voting key, which is not available on sentinel. Sentinel can only sign votes with the operator key.

### CODE REVIEW

File: src/governance-object.cpp [hunk 1]
```diff
@@ -790,3 +790,19 @@ void CGovernanceObject::CheckOrphanVotes(CConnman& connman)
...
+        std::string removedStr;
+        for (auto& h : removed) {
+            removedStr += strprintf("  %s\n", h.ToString());
+        }
+        LogPrintf("CGovernanceObject::RemoveOldVotes -- Removed %d old (pre-DIP3) votes for %s:\n%s\n", removed.size(), GetHash().ToString(), removedStr);
+        fDirtyCache = true;
+    }
+
+    return removed;
+}
```
Comment (UdjinM6) [inline src/governance-object.cpp]: newline at eof

File: src/governance.cpp [hunk 2]
```diff
@@ -672,6 +676,7 @@ void CGovernanceManager::SyncSingleObjAndItsVotes(CNode* pnode, const uint256& n
     pnode->PushInventory(CInv(MSG_GOVERNANCE_OBJECT, it->first));
 
     auto fileVotes = govobj.GetVoteFile();
+    bool onlyDIP3 = deterministicMNManager->IsDeterministicMNsSporkActive();
```
Comment (UdjinM6) [inline src/governance.cpp]: not used anywhere

File: src/governance.cpp [hunk 3]
```diff
@@ -1342,3 +1357,37 @@ void CGovernanceManager::CleanOrphanObjects()
         }
     }
 }
+
+unsigned int CGovernanceManager::GetMinVoteTime()
+{
+    LOCK(cs_main);
+    int64_t dip3SporkHeight = sporkManager.GetSporkValue(SPORK_15_DETERMINISTIC_MNS_ENABLED);
+    if (dip3SporkHeight == 4070908800ULL || chainActive.Height() < dip3SporkHeight) {
```
Comment (UdjinM6) [inline src/governance.cpp]:
```suggestion
    if (!deterministicMNManager->IsDeterministicMNsSporkActive()) {
```

File: src/governance-votedb.cpp [hunk 4]
```diff
@@ -68,6 +68,23 @@ void CGovernanceObjectVoteFile::RemoveVotesFromMasternode(const COutPoint& outpo
     }
 }
 
+std::vector<uint256> CGovernanceObjectVoteFile::RemoveOldVotes(unsigned int nMinTime)
+{
+    std::vector<uint256> removed;
+    vote_l_it it = listVotes.begin();
+    while (it != listVotes.end()) {
+        if (it->GetTimestamp() < nMinTime) {
+            --nMemoryVotes;
```
[the remark this answers is not in the document]
Comment (codablock) [reply src/governance-votedb.cpp]: Ahh I've only seen the mutex from CGovernanceManager and missed that there's also one for CGovernanceObject. Fixing this now.

**Pull request merged**